sos/report/plugins/mellanox_firmware.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (C) 2023 Nvidia Corporation, Alin Serdean <aserdean@nvidia.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, SoSPredicate, IndependentPlugin
+import os
+import time
+
+
+class MellanoxFirmware(Plugin, IndependentPlugin):
+
+    short_desc = 'Nvidia(Mellanox) firmware tools output'
+
+    plugin_name = "mellanox_firmware"
+    profiles = ('hardware', 'system')
+    commands = ('lspci -D -d 15b3::0200',)
```

Comment:
This is not what `commands` is for. This is used as an enablement trigger if the specified binary exists and is executable. E.G. `commands = ('mstconfig',)`  would enable this plugin only if that command exists (assuming no other enablement triggers are set).

Reply:
Just to make sure we are on the same page.  I do use it as an enablement trigger, enable plugin if and only if Mellanox network adapters are available.
I'm unsure if that's reasonable or not.